packages/client/src/features/tools/base-glsp-tool.ts

```diff
@@ -40,11 +47,11 @@ export abstract class BaseGLSPTool implements GLSPTool {
         this.actionDispatcher.dispatchAll(actions);
     }
 
-    dispatchFeedback(actions: Action[], feedbackEmitter?: IFeedbackEmitter): void {
-        this.feedbackDispatcher.registerFeedback(feedbackEmitter ?? this, actions);
+    registerFeedback(actions: Action[], feedbackEmitter?: IFeedbackEmitter, cleanupActions?: Action[]): Disposable {
+        return this.feedbackDispatcher.registerFeedback(feedbackEmitter ?? this, actions, cleanupActions);
     }
 
-    deregisterFeedback(actions: Action[], feedbackEmitter?: IFeedbackEmitter): void {
+    deregisterFeedback(actions?: Action[], feedbackEmitter?: IFeedbackEmitter): void {
```

Comment:
Parameter order should be reversed. Otherwise you have to call the method with undefined to deregister an emitter without actions. (`deregiserFeedback(undefined, someEmitter)`)
While we are at it we should also rename the `actions` parameter to `cleanupActions`

Reply:
Very nice! You are right, I'll fix it!